src/form/AsyncSelect/AsyncSelect.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { ComponentProps } from 'react';
+import { components, DropdownIndicatorProps } from 'react-select';
+import ReactSelectAsyncSelect from 'react-select/async';
+import { DropdownIndicatorStyle, getAsyncSelectStyle } from './AsyncSelectStyles';
+
+export type Props = {
+	error?: boolean;
+	zIndexMenu?: number;
+} & ComponentProps<typeof ReactSelectAsyncSelect>;
+
+// Todo: make this a reusable component, the typeahead above should be moved to AIP
+// Get possible streetnames by location https://api.data.amsterdam.nl/geosearch/bag/?lat=52.3912316&lon=4.954566&radius=100
+// Todo: what if API is down? -> show error message?
+// Todo: what if API is slow? -> show loading indicator?
+// Todo: what if API returns no results? -> show no results message?
+// Todo: what if API returns an error? -> show error message?
+// Todo: what if prefilled value is not available in API? -> show error message?
```

Comment:
Thats a bunch of todo's. So it doesn't do the API fetch yet?

Also perhaps we should add that it needs a react-hook-form controller to register this. 

I'm working with Mike on a spinner for the GPS component, we should look at adding loading state to the form? or Component ?

Reply:
The todo's are not meant for this component, this is a draft and these todo's are more or less questions / what ifs..